src/main/java/gregtech/common/covers/CoverRoboticArm.java

```diff
@@ -68,13 +68,15 @@ protected int doTransferExact(IItemHandler itemHandler, IItemHandler myItemHandl
         while (iterator.hasNext()) {
             TypeItemInfo sourceInfo = sourceItemAmount.get(iterator.next());
             int itemAmount = sourceInfo.totalCount;
-
-            // get the max we can extract from the item filter variable
             int itemToMoveAmount = itemFilterContainer.getSlotTransferLimit(sourceInfo.filterSlot);
-            int maxMultiplier = Math.floorDiv(Math.min(itemAmount, maxTransferAmount), itemToMoveAmount);
 
-            // multiply up to the total count of all the items
-            itemToMoveAmount *= Math.min(itemFilterContainer.getTransferStackSize(), maxMultiplier);
+            if (maxTransferAmount > 1) {
```

Comment:
I realize I made a mistake in my previous comment. This should actually check, `itemFilterContainer.getTransferStackSize() > 1`. This value is equal to 1 when no/the default multiplier is specified with a smart filter, so this calculation can be skipped in that case.